### PR TITLE
Fix store if_not_exists check error and add REMOTE database intergration tests

### LINKED ISSUE #454: Add integration tests
**Summary**
Add integration tests

### PR DESCRIPTION
**Pull request by BohuTANG**
## Summary

* Fix query&store `if_not_exists` check error
* Add REMOTE database/table create/insert intergration tests

## Changelog

- Bug Fix
- Build/Testing/CI

## Related Issues

Fixes #454 

## Test Plan

Unit Tests
Stateless Tests

### CODE REVIEW

**Review by dantengsky — COMMENTED** [review]
LGTM @drmingdrmer  any concerns?

File: fusestore/store/src/engine/mem_engine.rs [hunk 1]
```diff
@@ -38,8 +38,12 @@ impl MemEngine {
...
         // TODO: support plan.engine plan.options
         let curr = self.dbs.get(&cmd.db_name);
-        if curr.is_some() && if_not_exists {
-            return Err(anyhow::anyhow!("database exists"));
+        if let Some(curr) = curr {
+            return if if_not_exists {
+                Ok(curr.db_id)
+            } else {
+                Err(anyhow::anyhow!("{} database exists", cmd.db_name))
+            };
```
Comment (drmingdrmer) [inline fusestore/store/src/engine/mem_engine.rs:46]: If the DB exists, and `if_not_exists==true`, shouldn't it return an error? 
Without an error, the caller does not know whether a new DB is created or not.

And when `if_not_exists==false`, I think it means no matter the DB exists or not, it should always create a new DB but not returning an error. 🤔 ?

File: fusestore/store/src/engine/mem_engine.rs [hunk 2]
```diff
@@ -80,8 +84,12 @@ impl MemEngine {
...
             .get(&cmd.table_name);
 
-        if table_id.is_some() && if_not_exists {
-            return Err(Status::already_exists("table exists"));
+        if let Some(table_id) = table_id {
+            return if if_not_exists {
+                Ok(*table_id)
+            } else {
+                Err(Status::already_exists("table exists"))
+            };
```
Comment (drmingdrmer) [inline fusestore/store/src/engine/mem_engine.rs:92]: Same as the above question.

**Review by drmingdrmer — COMMENTED** [review]
Do not understand why the logic about `if_not_exists` is reversed

**Review by drmingdrmer — APPROVED** [review]

File: fusestore/store/src/engine/mem_engine.rs (hunk 1 above)
Comment (BohuTANG) [reply fusestore/store/src/engine/mem_engine.rs:46]: See:
https://dev.mysql.com/doc/refman/8.0/en/create-database.html
An error occurs if the database exists and you did not specify IF NOT EXISTS.

If we do `create db/table` and won't get an error, we can use it.

**Pull request merged**